Documentation/Network.md

```diff
@@ -93,22 +152,44 @@ In the same way that network requests are commonly made via HTTP, it's also very
+
+The `Payload` is the main source of data to perform the decoding, but on some scenarios an additional `Metadata` can be helpful (e.g. information contained in response headers).
 
-The fetching action of an HTTP network stack, in case of error, should throw an error of the [`Network.Error`][Network.Error] type. This type encapsulates the different possible error scenarios:
+By being a struct and not a protocol (while modelling the same behavior), it greatly simplifies generics and allows easy default implementations via static factory methods. It currently provides an `.json()` helper to build an `Decoding<T, Data, _>` that attempts to decode a particular `Decodable` model `T` encoded in JSON using a `JSONDecoder`.
+
+### Errors
+
+#### `URLSessionError`
+
+The fetching action of an HTTP network stack, in case of error, should throw an error of the [`URLSessionError`][URLSessionError] type. This type encapsulates the different possible error scenarios:
 
 * `noRequest`, when the resource's `makeRequest` fails.
-* `http`, when the request failed with an HTTP protocol error (i.e. non 2xx status code).
-* `api`, when the request failed with an HTTP protocol error (i.e. non 2xx status code), but a custom API error was produced by the resource's `decodeError` closure. The type of this error is the `Error` associated type of the resource which the stack is trying to fetch (provided via `ExternalErrorDecoderResource`).
+* `http`, when the request failed with an HTTP protocol error (i.e. non 2xx status code), and may contain a custom API error decoded by the resource's `errorDecoding` witness.
 * `noData`, when the response body is unexpectedly empty.
-* `url`, when the request fails with a network failure (e.g. the `error` in a dataTask's completion handler is non `nil`).
+* `url`, when the request fails with a network failure, expressed as an `URLError` (i.e. the `error` in a dataTask's completion handler is non `nil`).
 * `badResponse`, when a valid HTTP response is missing.
-* `retry`, when the request was not retried when evaluated by its retry policies after having failed with an error.
+* `retry`, when the request was _explicitly_ **not retried** when evaluated by its retry policies after having failed with an error.
+* `cancelled`, when the fetch was cancelled via the `Cancelable` instance.
+
+#### `FetchAndDecodeError`
+
+The [`FetchAndDecodeError`][FetchAndDecodeError] is a simple error type used on `fetchAndDecode` calls and is used to differentiate between errors originating from either the fetch or decode operations. As such, it has just two cases which wrap an error each:
+* `fetch(Error`
+* `decode(Error)`
 
 ## Usage
 
 ### Setup
 
+Let's walk through a basic steps required to start making some requests with Alicerce. A similar setup is also available in a [Swift playground][Network.xcplaygroundpage] as a live example.
```

Comment:
Should be something like `Let's walk through the basic steps`

Reply:
Indeed! will fix